docs/products/alloydb-omni/access-ai-models.md

```diff
@@ -0,0 +1,143 @@
+
+   ```sql
+   CREATE EXTENSION google_ml_integration VERSION '1.3';
+   ```
+
+1. Grant the service user access to utilize Vertex AI using the `gcloud` CLI:
+
+   ```bash
+   gcloud projects add-iam-policy-binding GOOGLE_CLOUD_PROJECT_NAME
+     --member="serviceAccount:GOOGLE_SERVICE_ACCOUNT_PRINCIPAL"
+     --role="roles/aiplatform.user"
+   ```
+
+   Replace the following:
+   - `GOOGLE_CLOUD_PROJECT_NAME` with the name of your Google Cloud project
+   - `GOOGLE_SERVICE_ACCOUNT_PRINCIPAL` with your Google service account principal,
+     for example `abc-vertex-ai-sa@johndoe-test.iam.gserviceaccount.com`.
+
+1. Define the AI model on your Aiven for AlloyDB Omni service using the `psql` CLI:
+
+   ```sql
+   CALL google_ml.create_model(
+       model_id => 'gemini-1.0-pro',
+       model_request_url => 'https://us-central1-aiplatform.googleapis.com/v1/projects/GOOGLE_CLOUD_PROJECT_NAME/locations/us-central1/publishers/google/models/gemini-1.0-pro:streamGenerateContent',
+       model_provider => 'google',
+       model_auth_type => 'alloydb_service_agent_iam');
+   ```
+
+   Replace `GOOGLE_CLOUD_PROJECT_NAME` with the name of your Google Cloud project,
+   for example `johndoe-test-pj`.
+
+1. Use the model by running:
+
+   ```sql
+   SELECT google_ml.predict_row(
+     model_id =>'gemini-1.0-pro',
+     request_body =>'{ "contents": { "role": "user", "parts": { "text": "What is Aiven?" }  }}');
+   ```
+
+## Use Hugging Face
+
+1. Install the `google_ml_integration` extension in your database using the `psql` CLI:
+
+   ```sql
+   CREATE EXTENSION google_ml_integration VERSION '1.3';
+   ```
+
+1. Record your [Hugging Face access token](https://huggingface.co/docs/hub/en/security-tokens)
+   as a secret in Google Secret Manager.
+
+1. Grant your Google service account access to the created secret using the `gcloud` CLI:
+
+   ```bash
+   gcloud secrets add-iam-policy-binding SECRET_ID
+     --member="serviceAccount:GOOGLE_SERVICE_ACCOUNT_PRINCIPAL"
+     --role="roles/secretmanager.secretAccessor"
+   ```
+
+   Replace the following:
+   - `SECRET_ID` with the ID of you newly created secret recorded in the Google Secret
+     Manager, for example `johndoe-test`
+   - `GOOGLE_SERVICE_ACCOUNT_PRINCIPAL` with your Google service account principal,
+     for example `abc-vertex-ai-sa@johndoe-test.iam.gserviceaccount.com`
+
+1. Declare the secret for the integration using the `psql` CLI:
+
+   ```sql
+   CALL google_ml.create_sm_secret(
+       secret_id=>'LOCAL_SECRET_ID',
+       secret_path=>'projects/GOOGLE_CLOUD_PROJECT_NAME/secrets/SECRET_ID/versions/1');
+   ```
+
+   Replace the following:
+   - `LOCAL_SECRET_ID` with the local identifier for your secret, for example `johndoe-token`
+   - `GOOGLE_CLOUD_PROJECT_NAME` with the name of your Google Cloud project where the
+     secret in stored, for example `johndoe-test-pj`
+   - `SECRET_ID` with the ID of the secret created in Google Secret Manager, for example
+     `johndoe-test`
```

Comment:
These steps apply to OpenAI and or Bring-your-own-model steps as well, albeit with different secrets, secret names and IDs. I'm not sure whether we can generalize this or whether just repeat altogether, though.